bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -1872,6 +1870,9 @@ private void internalConditionalExpandToLevel(Widget widget, int level,
 					}
 				}
 			}
+			if (widget instanceof Item it && data != null && !isExpandable(it, null, data)) {
+				return;
+			}
```

Comment:
@jannisCode something tells me this doesn't belong here :-) (https://github.com/vi-eclipse/Eclipse-Platform/issues/123)

Please revert the changes to this class

Reply:
done